soroban-sdk/src/env.rs

```diff
@@ -259,9 +259,9 @@ impl Env {
     ///
     /// Equivalent to `panic!`, but with an error value instead of a string.
     #[doc(hidden)]
+    #[inline(always)]
     pub fn panic_with_error(&self, error: impl Into<internal::Error>) {
         _ = internal::Env::fail_with_error(self, error.into());
-        unreachable!()
```

Comment:
As far as I can tell this unreachable is unnecessary. The reason we benefit from unreachable in the macro is so that we let the compiler know the code path will stop wherever the macro is used.

Reply:
Mm. I would actually think the content of the macro should move here and this function should be typed as `!` rather than `()`, but maybe there's some context where you want it to be typed this way? I don't really know why there's a macro at all tbh.

Reply:
Done in bfcc121. Thank you!